### DISCUSSION TITLE
What are these options used for ("Multiple", "Mark Elements", "Bubbles", "Trigger Event" & more)?

### DISCUSSION THREAD

**Discussion by ShlomoZee** [AutomaApp/automa#52] [Q&A]
I am not sure what the following options are capable of. 
Please describe what each of these can be used for:

• Multiple
• Mark Elements
• Bubbles
• Cancelable
• Trigger Event
• Code & Key (under "Trigger Event" > "Keypress" )
• Repeat (under "Trigger Event" > "Keypress" )
• Submit (under "Trigger Event")

**Comment by ABrandt91** [comment]
Same

**Comment by Kholid060** [comment]
### Multiple
Get multiple elements instead of one. Like when you're using the Get text block with `.paragraph` as the selector, instead of getting the text on one element it will get all the text from elements that match the selector.
Or if you know javascript, it will `querySelectorAll` instead of `querySelector`.

### Mark element
This will mark the element that means it will not be selected if it has been selected before by the same block. For example, when you have  **Get text block** with `.text` as the selector and the website that the workflow access has DOM tree like this
```HTML
<div>
  <p class="text" id="1">Text</p>
  <p class="text" id="2">Text</p>
  <p class="text" id="3">Text</p>
  <p class="text" id="4">Text</p>
</div>
```
When the **Get text block** is being executed, it get the first element that match the selector which is `<p class="text" id="1">Text</p>` and then it will mark it and it will look like this `<p class="text" id="1" block--block-id>Text</p>`. And in the DOM tree
```HTML
<div>
  <p class="text" id="1" block--block-id>Text</p>
  <p class="text" id="2">Text</p>
  <p class="text" id="3">Text</p>
  <p class="text" id="4">Text</p>
</div>
```
And in case after the **Get text block** you have **Repeat task block** and you connect the `repeat from` to the **Get text block**. When the **Get text block** being executed for the second time, It will do the same thing all over but instead select the `<p class="text" id="1" block--block-id>Text</p>` it will select the `<p class="text" id="2">Text</p>` element because the first element has been marked.

### Bubbles
This option is to tell whether the event bubbles or not. In javascript when an event is triggered in an element, it will execute the handler of the element first and then the parent of the element, and then all the way up. And that what bubbles is, for example, when a website have a DOM tree like this
```html
<div>
   <form>
      <input />
   </form>
</div>
```
when you trigger the `keyup` event into the `input` element using the trigger block, javascript will execute the input `keyup` handler at first, and then the `form` `keyup` event handler, and so on.

Read more at the [MDN page](https://developer.mozilla.org/en-US/docs/Web/API/Event/bubbles)

### Cancelable
> The cancelable read-only property of the Event interface indicates whether the event can be canceled, and therefore prevented as if the event never happened.

Read more about the cancelable at the [MDN page](https://developer.mozilla.org/en-US/docs/Web/API/Event/cancelable)

### Code & Key
Basically, code & key is property to tell which key is pressed. Like when you want to trigger `keyup` or `keydown` event with key `a` is being pressed, you can input `a` at the `key` and `KeyA` at the `code`.

> The `KeyboardEvent.code` property represents a physical key on the keyboard (as opposed to the character generated by pressing the key). In other words, this property returns a value that isn't altered by keyboard layout or the state of the modifier keys.

Read more about the `code` at the [MDN page](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/code)

> The KeyboardEvent interface's key read-only property returns the value of the key pressed by the user, taking into consideration the state of modifier keys such as Shift as well as the keyboard locale and layout.

Read more about `key` at the [MDN page](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key)

### Repeat
> The `repeat` read-only property of the KeyboardEvent interface returns a boolean value that is true if the given key is being held down such that it is automatically repeating.

Read more about `repeat` at the [MDN page](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/repeat)

### Submit
To trigger submit event in the [`form` element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/form).
